File: playwright-python/page_objects/login_page.py
# page_objects/login_page.py
# ─────────────────────────────────────────────────────────────────────────────
# Login Page Object - Locator constants only
# Rules: No logic | No imports | No methods | Grouped by section
# ─────────────────────────────────────────────────────────────────────────────

# ── Page identification ───────────────────────────────────────────────────────
PAGE_HEADING = "h1"
PAGE_TITLE = "title"

# ── Form inputs ───────────────────────────────────────────────────────────────
EMAIL_INPUT = "input[name='email']"
USERNAME_INPUT = "input[name='username']"
PASSWORD_INPUT = "input[name='password']"

# ── Buttons ───────────────────────────────────────────────────────────────────
LOGIN_BUTTON = "button[type='submit']"
SUBMIT_BUTTON = "//button[contains(., 'Login')]"
REGISTER_LINK = "a:has-text('Register')"

# ── Validation messages ───────────────────────────────────────────────────────
ERROR_MESSAGE = "[role='alert']"
VALIDATION_ERROR = ".error-message"
INVALID_CREDENTIALS_MSG = "//div[contains(text(), 'Invalid credentials')]"
MISSING_USERNAME_MSG = "//div[contains(text(), 'missing username')]"
MISSING_PASSWORD_MSG = "//div[contains(text(), 'missing password')]"
REQUIRED_FIELD_ERROR = "span:has-text('Required')"

# ── Success indicators ────────────────────────────────────────────────────────
SUCCESS_MESSAGE = ".success-message"
AUTH_TOKEN_RESPONSE = "[data-testid='auth-token']"

# ── Navigation ────────────────────────────────────────────────────────────────
DASHBOARD_INDICATOR = "[data-testid='dashboard']"
LOGOUT_BUTTON = "button:has-text('Logout')"

File: playwright-bdd-cucumber/pageObjects/loginPage.js
// pageObjects/loginPage.js
// ─────────────────────────────────────────────────────────────────────────────
// Feature: Login
// Generated: Qentrix ScriptGenerationAgent
// Selector Priority: id > role/accessibility > dynamic XPath > stable attributes > stable CSS > text
// ─────────────────────────────────────────────────────────────────────────────

// ── Page header / title ───────────────────────────────────────────────────────
export const pageHeading        = 'h1';
export const loginFormTitle     = 'h2:has-text("Login")';

// ── Form inputs ───────────────────────────────────────────────────────────────
export const emailInput         = 'input[name="email"]'; // TODO: verify selector against live app
export const usernameInput      = 'input[name="username"]'; // TODO: verify selector against live app
export const passwordInput      = 'input[type="password"]';

// ── Buttons ───────────────────────────────────────────────────────────────────
export const loginBtn           = 'button[type="submit"]';
export const registerLink       = 'a:has-text("Register")';

// ── Validation / error messages ───────────────────────────────────────────────
export const errorMessage       = '[role="alert"]'; // TODO: verify selector against live app
export const invalidCredentialsMsg = 'text="Invalid credentials"'; // TODO: verify selector against live app
export const requiredFieldError = 'span:has-text("Required")';
export const missingUsernameMsg = 'text="missing username"'; // TODO: verify selector against live app
export const missingPasswordMsg = 'text="missing password"'; // TODO: verify selector against live app

// ── Success indicators ────────────────────────────────────────────────────────
export const authToken          = '[data-testid="auth-token"]'; // TODO: verify selector against live app
export const dashboardUrl       = '/dashboard'; // TODO: verify selector against live app

